src/devices/Hx711/Hx711.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+
+namespace Iot.Device.Hx711
+{
+    /// <summary>
+    /// Controller for Hx711 - 24-bit Analog-to-Digital Converter for Weigh Scales
+    /// </summary>
+    /// <remarks>
+    /// Data sheet available here: https://cdn.sparkfun.com/datasheets/Sensors/ForceFlex/hx711_english.pdf
+    /// Based off ms-iot implementation here: https://github.com/ms-iot/hx711/blob/master/HX711.cs 
+    /// </remarks>
+    public class Hx711 : IDisposable
+    {
+        private readonly object _syncRoot = new object();
+
+        private readonly GpioController _gpioController;
+        private readonly int _dataPinNumber;
+        private readonly int _clockPinNumber;
+
+        private bool _enabled;
+
+        /// <summary>
+        /// Creates a new instance of the HC-SCR501.
+        /// </summary>
+        /// <param name="gpioController">The <see cref="GpioController"/> for the clock and data pins</param>
+        /// <param name="dataPin">The number of the pin to use for data</param>
+        /// <param name="clockPin">The number of the pin to use for clock</param>
+        /// <param name="pinNumberingScheme">Pin Numbering Scheme</param>
+        public Hx711(GpioController gpioController, int dataPin, int clockPin, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical)
+        {
+            _gpioController = gpioController;
+            _dataPinNumber = dataPin;
+            _clockPinNumber = clockPin;
+        }
+        
+        /// <summary>
+        /// Releases any resources
+        /// </summary>
+        public void Dispose()
+        {
+            Disable();
+        }
+
+        private byte ShiftInByte()
+        {
+            byte value = 0x00;
+
+            // Convert "GpioPinValue.High" and "GpioPinValue.Low" to 1 and 0, respectively.
+            // NOTE: Loop is unrolled for performance
+            _gpioController.Write(_clockPinNumber, PinValue.High);
+            value |= (byte)((byte)(_gpioController.Read(_dataPinNumber)) << 7);
+            _gpioController.Write(_clockPinNumber, PinValue.Low);
+            _gpioController.Write(_clockPinNumber, PinValue.High);
```

Comment:
How about this?

for (int i = 7; i == 0; i--)
{
_gpioController.Write(_clockPinNumber, PinValue.High);
value |= (byte)((byte)(_gpioController.Read(_dataPinNumber)) << i);
 _gpioController.Write(_clockPinNumber, PinValue.Low);
}

Reply:
Hi, thanks for the comment. This code is as provided by the ms-iot implementation. There is a note above it stating "Loop is unrolled for performance". Didn't really see any particular reason to change it but also have no strong feelings about it either.

Reply:
Haha, woops! Didn't read the comment! My bad, though I'm intrigued whether it actually matters given that the code is compiled.

Reply:
After doing some research, my guess is that the C# compiler will do the loop unrolling for you anyway, so you may as well have more maintainable code. This is not a definite answer though; I may be wrong.